File: src/main/java/com/example/oauth/config/properties/AuthorizationTokenProperties.java
package com.example.oauth.config.properties;

import com.example.oauth.model.token.TokenType;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.Setter;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;

@Component
@Getter(value = AccessLevel.PRIVATE)
@Setter
public class AuthorizationTokenProperties {
    private final String accessTokenCookieKey = "access-token";
    private final String refreshTokenCookieKey = "refresh-token";

    @Value("${security.token.access.secret}")
    private String accessTokenSecret;
    @Value("${security.token.access.expirationMsec}")
    private Long accessTokenExpirationMsec;

    @Value("${security.token.refresh.secret}")
    private String refreshTokenSecret;
    @Value("${security.token.refresh.expirationMsec}")
    private Long refreshTokenExpirationMsec;

    public String getTokenSecret(TokenType tokenType) {
        switch (tokenType) {
            case ACCESS_TOKEN:
                return this.getAccessTokenSecret();
            case REFRESH_TOKEN:
                return this.getRefreshTokenSecret();
            default:
                return "";
        }
    }

    public Long getTokenExpirationMsec(TokenType tokenType) {
        switch (tokenType) {
            case ACCESS_TOKEN:
                return this.getAccessTokenExpirationMsec();
            case REFRESH_TOKEN:
                return this.getRefreshTokenExpirationMsec();
            default:
                return 0L;
        }
    }

    public String getTokenCookieKey(TokenType tokenType) {
        switch (tokenType) {
            case ACCESS_TOKEN:
                return this.getAccessTokenCookieKey();
            case REFRESH_TOKEN:
                return this.getRefreshTokenCookieKey();
            default:
                return "";
        }
    }

    public int getTokenCookieMaxAge(TokenType tokenType) {
        switch (tokenType) {
            case ACCESS_TOKEN:
                return this.getAccessTokenCookieMaxAge();
            case REFRESH_TOKEN:
                return this.getRefreshTokenCookieMaxAge();
            default:
                return 0;
        }
    }

    private int getAccessTokenCookieMaxAge() {
        return Long.valueOf(accessTokenExpirationMsec / 1000).intValue();
    }

    private int getRefreshTokenCookieMaxAge() {
        return Long.valueOf(refreshTokenExpirationMsec / 1000).intValue();
    }
}

File: src/main/java/com/example/oauth/config/properties/AuthorizationTokenProperties.kt
package com.example.oauth.config.properties

import com.example.oauth.model.token.TokenType
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Component

@Component
class AuthorizationTokenProperties(
    private val accessTokenCookieKey: String = "access-token",
    private val refreshTokenCookieKey: String = "refresh-token",

    @Value("\${security.token.access.secret}")
    private val accessTokenSecret: String,

    @Value("\${security.token.access.expirationMsec}")
    private val accessTokenExpirationMsec: Long,

    @Value("\${security.token.refresh.secret}")
    private val refreshTokenSecret: String,

    @Value("\${security.token.refresh.expirationMsec}")
    private var refreshTokenExpirationMsec: Long,
) {

    fun tokenSecret(tokenType: TokenType): String {
        return when (tokenType) {
            TokenType.ACCESS_TOKEN -> accessTokenSecret
            TokenType.REFRESH_TOKEN -> refreshTokenSecret
        }
    }

    fun tokenExpirationMsec(tokenType: TokenType): Long {
        return when (tokenType) {
            TokenType.ACCESS_TOKEN -> accessTokenExpirationMsec
            TokenType.REFRESH_TOKEN -> refreshTokenExpirationMsec
        }
    }

    fun tokenCookieKey(tokenType: TokenType): String {
        return when (tokenType) {
            TokenType.ACCESS_TOKEN -> accessTokenCookieKey
            TokenType.REFRESH_TOKEN -> refreshTokenCookieKey
        }
    }

    fun tokenCookieMaxAge(tokenType: TokenType): Int {
        return when (tokenType) {
            TokenType.ACCESS_TOKEN -> (accessTokenExpirationMsec / 1000).toInt()
            TokenType.REFRESH_TOKEN -> (refreshTokenExpirationMsec / 1000).toInt()
        }
    }

}